src/components/environment/EnvironmentDetail.js

```diff
@@ -0,0 +1,207 @@
+import IconButton from '@material-ui/core/IconButton';
+import Collapse from '@material-ui/core/Collapse';
+import CardContent from '@material-ui/core/CardContent';
+
+import constants from '../../utils/constants';
+import EnvironmentDetailContent from './EnvironmentDetailContent';
+
+const styles = theme => ({
+  root: {
+    display: 'flex',
+    flexFlow: 'row wrap',
+  },
+  title: {
+    margin: theme.spacing.unit * 2,
+    marginTop: theme.spacing.unit * 6,
+  },
+  gridStyle: {
+    paddingTop: '50px',
+    backgroundColor: 'AliceBlue',
+  },
+  gridImg: {
+    width: '300px',
+    margin: theme.spacing.unit,
+    [theme.breakpoints.down('sm')]: {
+      width: '200px',
+    },
+  },
+  gridContent: {
+    width: '600px',
+    margin: theme.spacing.unit,
+    [theme.breakpoints.down('md')]: {
+      width: '400px',
+    },
+  },
+  imgCardStyle: {
+    width: '175px',
+    [theme.breakpoints.down('sm')]: {
+      width: '150px',
+    },
+  },
+  mediaStyle: {
+    height: '175px',
+    [theme.breakpoints.down('sm')]: {
+      height: '150px',
+    },
+  },
+
+  readmeCardContent: {
+    display: 'flex',
+    '&:last-child': {
+      paddingBottom: 0,
+    },
+  },
+  readmeStyle: {
+    margin: theme.spacing.unit,
+    textAlign: 'left',
+  },
+  expand: {
+    transform: 'rotate(0deg)',
+    transition: theme.transitions.create('transform', {
+      duration: theme.transitions.duration.shortest,
+    }),
+    position: 'relative',
+    left: '90%',
+    top: '-10px',
+    [theme.breakpoints.down('xs')]: {
+      left: '80%',
+    },
+  },
+  expandOpen: {
+    transform: 'rotate(180deg)',
+  },
+  readmePaperStyle: {
+    maxHeight: '500px',
+    overflow: 'scroll',
+    maxWidth: '700px',
+    align: 'center',
+    margin: 'auto',
+    textAlign: 'center',
+    [theme.breakpoints.down('xs')]: {
+      maxWidth: '320px',
+      maxHeight: '320px',
+    },
+  },
+  readmeCardStyle: {
+    marginTop: '40px',
+    margin: 'auto',
+    width: '90%',
+    [theme.breakpoints.up('md')]: {
+      width: '75%',
+    },
+    backgroundColor: 'AliceBlue',
+    paddingBottom: '0px',
+  },
+});
+
+class EnvironmentDetail extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      openReadme: false,
+    };
+  }
+
+  handleClickOpen = () => {
+    const newStateOpen = !this.state.openReadme;
+    this.setState({
+      openReadme: newStateOpen,
+    });
+  };
+
+  handleClose = () => {
+    this.setState({ openReadme: false });
+  };
+
+  render() {
+    const { classes, environment } = this.props;
+    var readme = '';
+    if (!(environment === undefined)) {
+      readme = this.props.environment.repository.readme;
+      const { currentAvatar } = this.props.environment;
+      var filePath = constants.SCIGYM_LOGO;
+      if (currentAvatar != null) {
+        filePath = currentAvatar.filePath.replace(constants.UPLOAD_URL, '');
+      }
+    }
+    var converter = new showdown.Converter();
```

Comment:
As is this will make a new converter every time anything changes in this render which seems like it could be a lot & often. I think that you can probably do this in the class' constructor rather than the render.

Reply:
done